service_management/azure/test/integration/service_bus/relay_test.rb

```diff
@@ -14,119 +14,141 @@
-    relay = subject.create_relay Azure::ServiceBus::Relay.new(name_alternative, description_alternative)
-    relay.must_be_kind_of Azure::ServiceBus::Relay
-    relay.name.must_equal name_alternative
+      relay.must_be_kind_of Azure::ServiceBus::Relay
+      relay.name.must_equal name_alternative
+      relay.relay_type.must_equal description_alternative[:relay_type]
+      relay.requires_client_authorization.must_equal description_alternative[:requires_client_authorization]
+      relay.requires_transport_security.must_equal description_alternative[:requires_transport_security]
+    end
+
+    it "should create a new relay from a relay with a description hash" do
+      relay = subject.create_relay Azure::ServiceBus::Relay.new(name_alternative, description_alternative)
 
-    relay.relay_type.must_equal description_alternative[:relay_type]
-    relay.requires_client_authorization.must_equal description_alternative[:requires_client_authorization]
-    relay.requires_transport_security.must_equal description_alternative[:requires_transport_security]
+      relay.must_be_kind_of Azure::ServiceBus::Relay
+      relay.name.must_equal name_alternative
+      relay.relay_type.must_equal description_alternative[:relay_type]
+      relay.requires_client_authorization.must_equal description_alternative[:requires_client_authorization]
+      relay.requires_transport_security.must_equal description_alternative[:requires_transport_security]
+    end
   end
 
-  describe 'when a relay exists' do
-    before { subject.create_relay bus_name, description }
+  describe "when a relay exists" do
+    before do
+      VCR.insert_cassette "service_bus/#{name}"
+      subject.create_relay relay_name, description
+    end
 
-    describe "#get_relay" do
-      it "should be able to get a relay by name" do
-        result = subject.get_relay bus_name
+    after do
+      subject.delete_relay relay_name
+      VCR.eject_cassette
+    end
 
-        result.must_be_kind_of Azure::ServiceBus::Relay
-        result.name.must_equal bus_name
-      end
+    it "should get a relay by name" do
+      result = subject.get_relay relay_name
 
-      it "if the relay endpoint doesn't exists it should throw" do
-        assert_raises(Azure::Core::Http::HTTPError) do
-          subject.get_relay ServiceBusRelayNameHelper.name
-        end
-      end
+      result.must_be_kind_of Azure::ServiceBus::Relay
+      result.name.must_equal relay_name
     end
 
-    describe '#delete_relay' do
-      it "should raise exception if the relay endpoint cannot be deleted" do
-        assert_raises(Azure::Core::Http::HTTPError) do
-          subject.delete_relay ServiceBusRelayNameHelper.name
-        end
+    it "should delete the relay endpoint" do
+      subject.create_relay name_alternative, description_alternative
+      response = subject.delete_relay name_alternative
+      response.must_equal nil
+    end
+
+    it "should throw if trying to get non existing relay" do
```

Comment:
should we take this and the next test out of this describe, to not create and delete a relay unnecessarily?